presto-main/src/main/java/com/facebook/presto/operator/scalar/MapToMapCast.java

```diff
@@ -213,25 +212,19 @@ public static Block mapCast(
             SqlFunctionProperties properties,
             Block fromMap)
     {
-        checkState(toMapType.getTypeParameters().size() == 2, "Expect two type parameters for toMapType");
-        Type toKeyType = toMapType.getTypeParameters().get(0);
-        TypedSet typedSet = new TypedSet(toKeyType, fromMap.getPositionCount() / 2, "map-to-map cast");
-        BlockBuilder keyBlockBuilder = toKeyType.createBlockBuilder(null, fromMap.getPositionCount() / 2);
-        for (int i = 0; i < fromMap.getPositionCount(); i += 2) {
-            try {
-                keyProcessFunction.invokeExact(fromMap, i, properties, keyBlockBuilder);
-            }
-            catch (Throwable t) {
-                throw internalError(t);
-            }
+        if (keyProcessFunction == null && valueProcessFunction == null) {
+            return fromMap;
```

Comment:
assert fromMap is not an instance of blockBuilder.

Reply:
We have this kind of usage in many scalar functions on arrays and maps so I don't want to add it here to avoid confusion.